### ISSUE TITLE
[CFE] Crash when variable type is not specified with clamp

### ISSUE THREAD

**Issue by letsar** [dart-lang/sdk#48812]
This code crashes on VM and dart2js
```dart
final a = 24.0.clamp(2.0, 30.0);

void main(){}
```

<details>
  <summary>StackTrace from DartPad</summary>

```
Error compiling to JavaScript:
Info: Compiling with sound null safety
Warning: Interpreting this as package URI, 'package:dartpad_sample/main.dart'.
lib/main.dart:
Internal Error: The compiler crashed when compiling this element.

The compiler is broken.

When compiling the above element, the compiler crashed. It is not
possible to tell if this is caused by a problem in your program or
not. Regardless, the compiler should not crash.

The Dart team would greatly appreciate if you would take a moment to
report this problem at http://dartbug.com/new.

Please include the following information:

* the name and version of your operating system,

* the Dart SDK build number (2.16.2), and

* the entire message you see here (including the full stack trace
  below as well as the source location above).

The compiler crashed: Crash when compiling null,
at character offset null:
Null check operator used on a null value
#0      TypeInferrerImpl._inferInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:2471:41)
#1      TypeInferrerImpl.inferInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:2114:12)
#2      TypeInferrerImpl._inferInstanceMethodInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:3226:40)
#3      TypeInferrerImpl.inferMethodInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:3761:20)
#4      InferenceVisitor.visitMethodInvocation (package:front_end/src/fasta/kernel/inference_visitor.dart:2893:21)
#5      MethodInvocation.acceptInference (package:front_end/src/fasta/kernel/internal_ast.dart:4611:20)
#6      TypeInferrerImpl._inferExpression (package:front_end/src/fasta/type_inference/type_inferrer.dart:1987:27)
#7      TypeInferrerImpl.inferExpression (package:front_end/src/fasta/type_inference/type_inferrer.dart:2024:40)
#8      _ImplicitFieldTypeRoot.computeType (package:front_end/src/fasta/kernel/implicit_field_type.dart:157:55)
#9      SourceFieldBuilder.inferType (package:front_end/src/fasta/builder/field_builder.dart:503:47)
#10     SourceLoader.performTopLevelInference (package:front_end/src/fasta/source/source_loader.dart:1962:35)
#11     KernelTarget.buildOutlines.<anonymous closure> (package:front_end/src/fasta/kernel/kernel_target.dart:411:14)
<asynchronous suspension>
#12     withCrashReporting (package:front_end/src/fasta/crash.dart:122:12)
<asynchronous suspension>
#13     generateKernelInternal.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:100:10)
<asynchronous suspension>
#14     withCrashReporting (package:front_end/src/fasta/crash.dart:122:12)
<asynchronous suspension>
#15     compile.<anonymous closure> (package:front_end/src/api_unstable/dart2js.dart:202:37)
<asynchronous suspension>
#16     compile (package:front_end/src/api_unstable/dart2js.dart:200:36)
<asynchronous suspension>
#17     KernelLoaderTask.load.<anonymous closure> (package:compiler/src/kernel/loader.dart:217:21)
<asynchronous suspension>
#18     Compiler.runInternal (package:compiler/src/compiler.dart:325:29)
<asynchronous suspension>
#19     main (package:compiler/src/dart2js.dart:1373:3)
<asynchronous suspension>


#0      TypeInferrerImpl._inferInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:2471:41)
#1      TypeInferrerImpl.inferInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:2114:12)
#2      TypeInferrerImpl._inferInstanceMethodInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:3226:40)
#3      TypeInferrerImpl.inferMethodInvocation (package:front_end/src/fasta/type_inference/type_inferrer.dart:3761:20)
#4      InferenceVisitor.visitMethodInvocation (package:front_end/src/fasta/kernel/inference_visitor.dart:2893:21)
#5      MethodInvocation.acceptInference (package:front_end/src/fasta/kernel/internal_ast.dart:4611:20)
#6      TypeInferrerImpl._inferExpression (package:front_end/src/fasta/type_inference/type_inferrer.dart:1987:27)
#7      TypeInferrerImpl.inferExpression (package:front_end/src/fasta/type_inference/type_inferrer.dart:2024:40)
#8      _ImplicitFieldTypeRoot.computeType (package:front_end/src/fasta/kernel/implicit_field_type.dart:157:55)
#9      SourceFieldBuilder.inferType (package:front_end/src/fasta/builder/field_builder.dart:503:47)
#10     SourceLoader.performTopLevelInference (package:front_end/src/fasta/source/source_loader.dart:1962:35)
#11     KernelTarget.buildOutlines.<anonymous closure> (package:front_end/src/fasta/kernel/kernel_target.dart:411:14)
<asynchronous suspension>
#12     withCrashReporting (package:front_end/src/fasta/crash.dart:122:12)
<asynchronous suspension>
#13     generateKernelInternal.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:100:10)
<asynchronous suspension>
#14     withCrashReporting (package:front_end/src/fasta/crash.dart:122:12)
<asynchronous suspension>
#15     compile.<anonymous closure> (package:front_end/src/api_unstable/dart2js.dart:202:37)
<asynchronous suspension>
#16     compile (package:front_end/src/api_unstable/dart2js.dart:200:36)
<asynchronous suspension>
#17     KernelLoaderTask.load.<anonymous closure> (package:compiler/src/kernel/loader.dart:217:21)
<asynchronous suspension>
#18     Compiler.runInternal (package:compiler/src/compiler.dart:325:29)
<asynchronous suspension>
#19     main (package:compiler/src/dart2js.dart:1373:3)
<asynchronous suspension>
```

</details>

But if we set the type of a, it doesn't:
```dart
final double a = 24.0.clamp(2.0, 30.0);

void main(){}
```

**Comment by renancaraujo** [comment]
Even though this is closed, the error can still be observed on 2.17.1 (stable).

**Comment by jensjoha** [comment]
Yes, this commit has not gone into stable.
From the tags overview on the commit it's in from tag `2.18.0-36.0.dev`.
